### PR TITLE
[MRG] Group lasso branch

### PR DESCRIPTION
**Pull request by themantalope**
added in requested fixes from the last PR, may still need to fix some flake8 errors

### CODE REVIEW

**Comment by pavanramkumar** [comment]
@themantalope thanks for ironing out all the circle CI issues. i just took a look. right now, you only report the score on the entire training set, but what you should ideally do is fit the model on a training set and report on a test set.

have a look at the other examples. if you are able to do this, i can merge it soon. thanks!

**Comment by themantalope** [comment]
Oh yeah, that's a quick fix, wasn't thinking :P

**Comment by pavanramkumar** [comment]
closes #154 #170 #174

**Comment by pavanramkumar** [comment]
i like the `tempfile` solution, will use it for other examples as well. thanks for the nice work! merging.

**Comment by pavanramkumar** [comment]
spoke too soon! there are some conflicts with `pyglmnet/datasets.py`. let me check

**Comment by pavanramkumar** [comment]
@themantalope i merged the conflicts but can you give me access to your `pyglmnet` fork so that i can push to `group-lasso-branch` branch and directly update the PR? you'd need to go to settings and add me as a collaborator. thanks!

**Comment by themantalope** [comment]
Also, just out of curiosity, what are the conflicts? I can see the warning message but I don't know where to go to see exactly what they are.
